src/gui/properties/propertieswidget.cpp

```diff
@@ -322,8 +322,14 @@ void PropertiesWidget::loadTorrentInfos(BitTorrent::TorrentHandle *const torrent
 
         // List files in torrent
         m_propListModel->model()->setupModelData(m_torrent->info());
-        if (m_propListModel->model()->rowCount() == 1)
-            m_ui->filesList->setExpanded(m_propListModel->index(0, 0), true);
+        if (m_propListModel->model()->rowCount() == 1) {
+            QModelIndex currentIndex = m_propListModel->index(0, 0);
+            // Expand single-item folders recursively
+            do {
+                m_ui->filesList->setExpanded(currentIndex, true);
+                currentIndex = m_propListModel->index(0, 0, currentIndex);
+            } while (m_propListModel->rowCount(currentIndex) == 1);
```

Comment:
Also will it loop forever in the case of a single-file torrent (without any folder structure)?

Reply:
hm, now I'm stuck. the same forever-loop should happen on a single-item folder, which has single file on the lowest level (see screenshot in #11525).
TorrentContentModel::index(0,0, last-file/single-file_like_you_said) should return {}, because 
`    if (row >= parentItem->childCount())
        return {};`
and then `TorrentContentModel::rowCount(currentIndex)` becomes `rowCount({})` and should return the row count of the `m_rootItem->childCount()`, so looping again.
but strangely I don't see CPU busy, nor lags in UI redrawing.
So I guess I don't understand something in the codebase. Please advise.

Reply:
```c++
QModelIndex currentIndex;
while (m_propListModel->rowCount(currentIndex) == 1) {
    m_ui->filesList->setExpanded(currentIndex, true);
    currentIndex = m_propListModel->index(0, 0, currentIndex);
}
```

Reply:
replaced the whole if() with while(), how about now?